components/lib/datatable/DataTable.js

```diff
@@ -714,6 +716,36 @@ export const DataTable = React.forwardRef((inProps, ref) => {
                 let isSameColumn = (col1, col2) => (getColumnProp(col1, 'columnKey') || getColumnProp(col2, 'columnKey') ? ObjectUtils.equals(col1.props, col2.props, 'columnKey') : ObjectUtils.equals(col1.props, col2.props, 'field'));
                 let dragColIndex = columns.findIndex((child) => isSameColumn(child, draggedColumn.current));
                 let dropColIndex = columns.findIndex((child) => isSameColumn(child, column));
+                let widths = [];
+                let headers = DomHandler.find(tableRef.current, '.p-datatable-thead > tr > th');
+                
+                headers.forEach((header) => widths.push(DomHandler.getOuterWidth(header)));
+                let selector = `.p-datatable[${attributeSelectorState}] > .p-datatable-wrapper ${isVirtualScrollerDisabled() ? '' : '> .p-virtualscroller'} > .p-datatable-table`;
+                const movedItem = widths.find((items, index) => index === dragColIndex);
+                const remainingItems = widths.filter((items, index) => index !== dragColIndex);
+                const reorderedWidths = [
+                    ...remainingItems.slice(0, dropColIndex),
+                    movedItem,
+                    ...remainingItems.slice(dropColIndex)
+                ];
+                let innerHTML = '';
+
+                destroyStyleElement();
+                createStyleElement();
+
+                reorderedWidths.forEach((width, index) => {
```

Comment:
I feel like this could be refactored into a method since it looks like `restoreColumnWidths` does almost the exact same thing?

Reply:
Hey @melloware, I've refactored the code as per your comment. Please review it.